core/src/main/java/org/pentaho/platform/util/oauth/PentahoOAuthUtility.java

```diff
@@ -0,0 +1,166 @@
+    return instance;
+  }
+
+  public static boolean isOAuthEnabled() {
+
+    IConfiguration config = systemConfig.getConfiguration( "security" );
+
+    if ( !isProviderOAuth ) {
+      try {
+        String provider = config.getProperties().getProperty( "provider" );
+        isProviderOAuth = StringUtils.equals( provider, OAUTH );
+        isProviderDualAuth = StringUtils.equals( provider, DUALAUTH );
+      } catch ( IOException e ) {
+        isProviderOAuth = false;
+      }
+    }
+
+    if ( !isProviderOAuth && isProviderDualAuth ) {
+      config = systemConfig.getConfiguration( OAUTH );
+      try {
+        String provider1 = config.getProperties().getProperty( "provider1" );
+        String provider2 = config.getProperties().getProperty( "provider2" );
+        isProviderOAuth = StringUtils.equals( provider1, OAUTH ) || StringUtils.equals( provider2, OAUTH );
+      } catch ( IOException e ) {
+        isProviderOAuth = false;
+      }
+    }
+
+    return isProviderOAuth;
+  }
+
+  public static boolean isOAuthEnabledWithDualAuth() {
+
+    IConfiguration config = systemConfig.getConfiguration( "security" );
+
+    if ( !isProviderDualAuth ) {
+      String provider = null;
+      try {
+        provider = config.getProperties().getProperty( "provider" );
+      } catch ( IOException e ) {
+        isProviderDualAuth = false;
+      }
+      isProviderDualAuth = StringUtils.equals( provider, DUALAUTH );
+    }
+
+    if ( isProviderDualAuth ) {
+      config = systemConfig.getConfiguration( DUALAUTH );
+      String provider1 = null;
+      String provider2 = null;
+      try {
+        provider1 = config.getProperties().getProperty( "provider1" );
+        provider2 = config.getProperties().getProperty( "provider2" );
+      } catch ( IOException e ) {
+        isProviderOAuth = false;
+      }
+      isProviderOAuth = StringUtils.equals( provider1, OAUTH ) || StringUtils.equals( provider2, OAUTH );
+    }
+
+    return isProviderOAuth;
+  }
+
+  public static boolean shouldPerformLiveUpdate() {
+    IConfiguration config = systemConfig.getConfiguration( OAUTH );
+    try {
+      return Boolean.parseBoolean( config.getProperties().getProperty( "shouldPerformLiveUpdate", "false" ) );
+    } catch ( IOException e ) {
+      return false;
+    }
+  }
+
+  public static boolean isUserNamePasswordAuthentication( HttpServletRequest httpServletRequest ) {
+    return "/pentaho/j_spring_security_check".equals( httpServletRequest.getRequestURI() ) ||
+      SecurityContextHolder.getContext().getAuthentication() instanceof UsernamePasswordAuthenticationToken;
+  }
+
+  public static String getUserNameAttribute( String registrationId ) {
+    IConfiguration config = systemConfig.getConfiguration( OAUTH );
+    try {
+      return config.getProperties().getProperty( registrationId + ".user-name-attribute" );
+    } catch ( IOException e ) {
+      return null;
+    }
+  }
+
+  public <T> ResponseEntity<T> getResponseEntity( String key,
```

Comment:
I don't understand this method, but it looks sketchy. Are we sure this is safe?

Reply:
there are some getEntity calls to be made in the PentahoOAuth*Handler.java classes. Each have different type of response types like Map, String, List etc., 

Hence, made it accepts <T> from the caller and returns generic response type <T> to the invoker.